ocaml/libs/http-lib/http_svr.ml

```diff
@@ -221,17 +221,23 @@ let response_redirect ?req s dest =
   in
   Unixext.really_write_string s (Http.Response.to_wire_string res)
 
-let response_file ?mime_content_type s file =
+let response_file ?mime_content_type ~add_hsts s file =
```

Comment:
Would it not be better to use an option that includes the value actually being used? So either `None` or `Some time`? The time to use could still come from a configuration.

Reply:
Right. Common values are 1yr and 2yr but it is a good idea to have something that is configurable. If I understand correctly I could make it optional and change the configuration to be **hsts_max_age** and set it to 0 by default. And it also remove the magic constant ;-)... I will do that. Thanks for the comment.

Reply:
Note that set it to 0 by default is to be able to not activate this feature. So if it is 0 I will not pass the value to response_file otherwise I will pass it. We want to disable it by default because HSTS prevent you to use self signed certificate and it is common to have self signed certificate. So only people that know what they are doing (and so have a valid configuration) should enable it.

Reply:
This is my idea, roughly. I did not implement the other comment about magic numbers,

```
diff --git a/ocaml/libs/http-lib/http_svr.ml b/ocaml/libs/http-lib/http_svr.ml
index ea35e02ee..b4ef6600c 100644
--- a/ocaml/libs/http-lib/http_svr.ml
+++ b/ocaml/libs/http-lib/http_svr.ml
@@ -221,17 +221,22 @@ let response_redirect ?req s dest =
   in
   Unixext.really_write_string s (Http.Response.to_wire_string res)
 
-let response_file ?mime_content_type s file =
+let response_file ?mime_content_type ?hsts s file =
   let size = (Unix.LargeFile.stat file).Unix.LargeFile.st_size in
   let mime_header =
     Option.fold ~none:[]
       ~some:(fun ty -> [(Hdr.content_type, ty)])
       mime_content_type
   in
-  let keep_alive = (Http.Hdr.connection, "keep-alive") in
+  let keep_alive = [(Http.Hdr.connection, "keep-alive")] in
+  let hsts =
+    Option.fold ~none:[]
+      ~some:(fun time -> [(Http.Hdr.hsts, "max-age=63072000")])
+      hsts
+  in
   let res =
-    Http.Response.make ~version:"1.1"
-      ~headers:(keep_alive :: mime_header)
+    Http.Response.make ~version:"1.1" ~headers:List.concat
+      [keep_alive; hsts; mime_header]
       ~length:size "200" "OK"
   in
   Unixext.with_file file [Unix.O_RDONLY] 0 (fun f ->
diff --git a/ocaml/libs/http-lib/http_svr.mli b/ocaml/libs/http-lib/http_svr.mli
index d20ddfc38..25a12c4f6 100644
--- a/ocaml/libs/http-lib/http_svr.mli
+++ b/ocaml/libs/http-lib/http_svr.mli
@@ -120,7 +120,7 @@ val response_method_not_implemented :
 val response_redirect : ?req:Http.Request.t -> Unix.file_descr -> string -> unit
 
 val response_file :
-  ?mime_content_type:string -> Unix.file_descr -> string -> unit
+  ?mime_content_type:string -> ?hsts:int -> Unix.file_descr -> string -> unit
 
 val respond_to_options : Http.Request.t -> Unix.file_descr -> unit
 
```